keras_cv/models/generative/stable_diffusion/stable_diffusion.py

```diff
@@ -127,27 +138,33 @@ def text_to_image(
             )
         phrase = inputs + [49407] * (MAX_PROMPT_LENGTH - len(inputs))
         phrase = tf.convert_to_tensor([phrase], dtype=tf.int32)
-        phrase = tf.repeat(phrase, batch_size, axis=0)
 
-        # Encode prompt tokens + positions into a "context" vector
-        pos_ids = tf.convert_to_tensor([list(range(MAX_PROMPT_LENGTH))], dtype=tf.int32)
-        pos_ids = tf.repeat(pos_ids, batch_size, axis=0)
-        context = self.text_encoder.predict_on_batch([phrase, pos_ids])
+        context = self.text_encoder.predict_on_batch([phrase, self._get_pos_ids()])
 
-        # Encode unconditional tokens + positions as "unconditional context"
-        unconditional_tokens = tf.convert_to_tensor(
-            [_UNCONDITIONAL_TOKENS], dtype=tf.int32
-        )
-        self.unconditional_tokens = tf.repeat(unconditional_tokens, batch_size, axis=0)
-        unconditional_context = self.text_encoder.predict_on_batch(
-            [self.unconditional_tokens, pos_ids]
+        return context
+
+    def generate_image(
+        self,
+        encoded_text,
+        batch_size=1,
+        num_steps=25,
+        unconditional_guidance_scale=7.5,
+        diffusion_noise=None,
+        seed=None,
+    ):
+        context = tf.repeat(encoded_text, batch_size, axis=0)
+        unconditional_context = tf.repeat(
+            self._get_unconditional_context(), batch_size, axis=0
         )
 
+        if diffusion_noise:
+            latent = diffusion_noise
+        else:
```

Comment:
should we error if a user passes both seed and diffusion noise?  If I understand correctly you pas one or the other.

Reply:
Yes, good call. I've updated this to throw an error and added a test (which I've manually run but subsequently added the DISABLED_ prefix to so that it doesn't get run)